src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -159,7 +176,7 @@ private void export(PeakList peakList, FileWriter writer, File curFile)
                 
                 String mass = Double.toString(row.getAverageMZ());
                 if (mass != null) writer.write("PEPMASS=" + mass + newLine);                                                                      
-                writer.write("CHARGE=1"+newLine);
+                writer.write("CHARGE="+row.getRowCharge()+"+"+newLine);
```

Comment:
Yes, thanks for seing that. This is correct. We can make the approximation that most of detected ions that are not deconvoluted are probably monocharged.
However, here we dont consider the ionisation mode here. We just put + arbitrarily.
We should have CHARGE=1+ or CHARGE=1- depending if it's positive or negative ion mode. 
Please let us know if you have suggestions on how to make that possible.